src/config_attribute_value.hpp

```diff
@@ -148,6 +148,7 @@ class config_attribute_value
 	std::time_t to_time_t(std::time_t def = 0) const;
 	double to_double(double def = 0.) const;
 	std::string str(const std::string& fallback = "") const;
+	std::string to_string(const std::string& fallback = "") const {return str(fallback);};
```

Comment:
I guess this exists for the purpose of preprocessor string concatenation?

Reply:
Yep. Same for the `using std::string;` in preferences.hpp.

Reply:
It would imo be better to have a template `to<T>(const T& fallback)` function (specialized for int, std::string, bool etc), this would not only make it unnecessary to have `using std::string` in perferences.hpp but would likely also be handy in other places of the code.

Reply:
That doesn't really help here though since I can't make a `to_std::string()` method.

Reply:
It would become `to<std::string>()`

Reply:
implemented like (untested code)
```
	template<typename T>
	T to(const T& def) const
	{
		if constexpr(std::is_same_v<T, std::string>)) {
			return str(def);
		}
		if constexpr(std::is_same_v<T, int>)) {
			return to_int(def);
		}
		if constexpr(std::is_same_v<T, bool>)) {
			return to_bool(def);
		}
        //Not sure whetehr static_assert works liek this
        static_assert(false, "unsupported type");
	}
```

And used like `return preferences_[prefs_list::pref].to<type>(def);`

Reply:
Added.

Reply:
Would it be cleaner to just define `to` as three overloaded functions in this case?

Reply:
I have no particular preference either way.

Reply:
Updated.

Reply:
> I have no particular preference either way.

I'm also fine with both